core/adapters/http.go

```diff
@@ -102,3 +105,63 @@ func sendRequest(input models.RunResult, request *http.Request) models.RunResult
 	input.ApplyResult(responseBody)
 	return input
 }
+
+// maxBytesReader is inspired by
+// https://github.com/gin-contrib/size/blob/master/size.go
+type maxBytesReader struct {
+	rc               io.ReadCloser
+	limit, remaining int64
+	sawEOF           bool
+}
+
+func newMaxBytesReader(rc io.ReadCloser, limit int64) *maxBytesReader {
+	return &maxBytesReader{
+		rc:        rc,
+		limit:     limit,
+		remaining: limit,
+	}
+}
+
+func (mbr *maxBytesReader) Read(p []byte) (n int, err error) {
+	toRead := mbr.remaining
+	if mbr.remaining == 0 {
+		if mbr.sawEOF {
+			return mbr.tooLarge()
+		}
+		// The underlying io.Reader may not return (0, io.EOF)
+		// at EOF if the requested size is 0, so read 1 byte
+		// instead. The io.Reader docs are a bit ambiguous
+		// about the return value of Read when 0 bytes are
+		// requested, and {bytes,strings}.Reader gets it wrong
+		// too (it returns (0, nil) even at EOF).
+		toRead = 1
+	}
+	if int64(len(p)) > toRead {
```

Comment:
Would it be better to use a uint64 here for `toRead` and `remaining` to avoid an overflow?